src/api/app/models/package.rb

```diff
@@ -308,12 +308,14 @@ def is_locked?
     project.is_locked?
   end
 
+  def master_product_object
+    # test _product permissions if any other _product: subcontainer is used and _product exists
+    product_object = project.packages.find_by(name: "_product") if name =~ /\A_product:\w[-+\w\.]*\z/
```

Comment:
these package names are auto-generated, you can't really change them.

the product definition is inside a package object. It is not different on the package meta level (except  that generating package must be named "_product"). The product itself is defined in source files inside. We have also product objects, but these are only indexed data following from the source.